python/python/lancedb/table.py

```diff
@@ -1889,6 +1908,23 @@ def compact_files(self, *args, **kwargs):
         """
         return self.to_lance().optimize.compact_files(*args, **kwargs)
 
+    def optimize_indices(self, **kwargs):
+        try:
+            asyncio.get_running_loop()
+            return self.to_lance().optimize.optimize_indices(**kwargs)
+        except RuntimeError:
+            result = asyncio.run(self._async_optimize_indices(**kwargs))
+            # need to checkout to the latest version to make the changes visible
+            self.checkout_latest()
+            return result
+
+    async def _async_optimize_indices(self, **kwargs):
+        conn = await lancedb_connect(
+            sanitize_uri(self._conn.uri),
+        )
+        table = AsyncTable(await conn.open_table(self.name))
+        return await table.optimize()
+
```

Comment:
This will be misleading.  `AsyncTable::optimize` does more than just `optimize_indices`.  It also runs compaction and cleans up older versions.  Can you change the sync method name to `optimize` and change the method description to be the same as `AsyncTable::optimize`:

```
    def optimize(
        self,
        *,
        cleanup_older_than: Optional[timedelta] = None,
        delete_unverified: bool = False,
    ) -> OptimizeStats:
        """
        Optimize the on-disk data and indices for better performance.

        Modeled after ``VACUUM`` in PostgreSQL.

        Optimization covers three operations:

         * Compaction: Merges small files into larger ones
         * Prune: Removes old versions of the dataset
         * Index: Optimizes the indices, adding new data to existing indices

        Parameters
        ----------
        cleanup_older_than: timedelta, optional default 7 days
            All files belonging to versions older than this will be removed.  Set
            to 0 days to remove all versions except the latest.  The latest version
            is never removed.
        delete_unverified: bool, default False
            Files leftover from a failed transaction may appear to be part of an
            in-progress operation (e.g. appending new data) and these files will not
            be deleted unless they are at least 7 days old. If delete_unverified is True
            then these files will be deleted regardless of their age.

        Experimental API
        ----------------

        The optimization process is undergoing active development and may change.
        Our goal with these changes is to improve the performance of optimization and
        reduce the complexity.

        That being said, it is essential today to run optimize if you want the best
        performance.  It should be stable and safe to use in production, but it our
        hope that the API may be simplified (or not even need to be called) in the
        future.

        The frequency an application shoudl call optimize is based on the frequency of
        data modifications.  If data is frequently added, deleted, or updated then
        optimize should be run frequently.  A good rule of thumb is to run optimize if
        you have added or modified 100,000 or more records or run more than 20 data
        modification operations.
        """
    ...
```

Reply:
ok, do we still need to make exsiting `compact_files` and `cleanup_old_versions` based on async impl?

Reply:
> ok, do we still need to make exsiting compact_files and cleanup_old_versions based on async impl?

No, we can leave those as sync-only and eventually deprecate.